packages/web/src/components/range/RangeInput.js

```diff
@@ -64,29 +58,34 @@ class RangeInput extends Component {
 			}
 		}
 
-		const currentValue = {
-			start: this.state.start,
-			end: this.state.end,
-			[name]: value,
-		};
-
-		const { value: valueProp, onChange } = this.props;
-		if (valueProp === undefined) {
+		if (this.isControlled) {
+			this.props.onChange({
+				...this.value,
+				[name]: value,
+			});
+		} else {
 			this.setState({
 				[name]: value,
 			});
-		} else if (onChange) {
-			onChange(currentValue);
 		}
 	};
 
-	handleSliderChange = ([start, end]) => {
-		const { value, onChange } = this.props;
+	get isControlled() {
+		return this.props.onChange;
```

Comment:
A component will be controlled when we have both `value` & `onChange` prop

Reply:
easy fix, I'll do that.

Do you think it's worth having a warning of some sort if only one of the two is supplied?

Reply:
Indeed, warnings will be a nice add-on. I believe we can add that validation via prop-types ~ maybe send in a separate PR for this if you wish to take this up :)

---

`value` prop is the factor that makes a controlled component. If `onChange` is absent when `value` prop is set, we can show a warning in the console.

Reply:
ah, cool. I can make it so `value` drives the `isControlled` functionality.

How can we use proptypes to conditionally require `onChange` based on whether `value` has been supplied? I don't think I've ever done anything like that before.

Reply:
You can add a custom validator for the desired prop. It's pretty interesting 😄 

Check the docs here: https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes

Reply:
oh cool, good know! Maybe I'll do this in another PR :)